tests/rptest/clients/ping_pong.py

```diff
@@ -0,0 +1,148 @@
+            "isolation.level": "read_committed"
+        })
+        self.consumer.assign(
+            [TopicPartition(topic, partition, OFFSET_BEGINNING)])
+        self.stream = self.stream_gen()
+
+    def stream_gen(self):
+        while True:
+            msgs = self.consumer.consume(timeout=10)
+            for msg in msgs:
+                yield msg
+
+    def read_until(self, check, timeout_s):
+        begin = time.time()
+        while True:
+            if time.time() - begin > timeout_s:
+                raise KafkaException(KafkaError(KafkaError._TIMED_OUT))
+            for msg in self.stream:
+                offset = msg.offset()
+                value = msg.value().decode('utf-8')
+                key = msg.key().decode('utf-8')
+                if check(offset, key, value):
+                    return
+
+
+def expect(offset, key, value):
+    def check(ro, rk, rv):
+        if ro < offset:
+            return False
+        if ro == offset:
+            if rk != key:
+                raise RuntimeError(f"expected key='{key}' got '{rk}'")
+            if rv != value:
+                raise RuntimeError(f"expected value='{value}' got '{rv}'")
+            return True
+        raise RuntimeError(f"read offset={ro} but skipped {offset}")
+
+    return check
+
+
+class PingPong:
+    def __init__(self, brokers, topic, partition, logger):
+        self.brokers = brokers
+        random.shuffle(self.brokers)
+        bootstrap = ",".join(self.brokers)
+        self.consumer = LogReader(bootstrap)
+        self.consumer.init(group="ping_ponger1",
+                           topic=topic,
+                           partition=partition)
+        self.producer = SyncProducer(bootstrap)
+        self.producer.init()
+        self.logger = logger
+        self.topic = topic
+        self.partition = partition
+
+    def ping_pong(self, timeout_s=5, retries=0):
+        key = str(random.randint(0, 1000))
+        value = str(random.randint(0, 1000))
+
+        start = time.time()
+
+        offset = None
+        count = 0
+        while True:
+            count += 1
+            try:
+                offset = self.producer.produce(topic=self.topic,
+                                               partition=self.partition,
+                                               key=key,
+                                               value=value,
+                                               timeout_s=timeout_s)
+                break
+            except KafkaException as e:
+                if count > retries:
+                    raise
+                if e.args[0].code() == KafkaError._MSG_TIMED_OUT:
+                    pass
+                elif e.args[0].code() == KafkaError._TIMED_OUT:
+                    pass
+                else:
+                    raise
+                random.shuffle(self.brokers)
+                bootstrap = ",".join(self.brokers)
+                self.producer = SyncProducer(bootstrap)
```

Comment:
I propose that we pause this PR until https://github.com/confluentinc/confluent-kafka-python/issues/1335 is resolved -- this test then becomes a good regression test for that bug, once we remove this client-rebuilding workaround.

Reply:
We may run into the same behavior even when they fix it. Also we may run into it with franz-go, kafka java client or others. There isn't clear spec for what the client should do if it runs into a retry-able error during init_producer_id: most clients choose a broker randomly and retry, some picks the least loaded and some retry with the same.

If we assume that the clients should choose at random then it is still a probability game and we either should set big enough timeout to let it guess the right node or to retry on the app side. And if we set huge timeout it may mask other problems!

Ok, it isn't a problem with `confluent_kafka` because it supports per request timeout but java client doesn't. So if I rewrite the test now to use less buggy client to avoid the retry on the app level then I'll still need to have it for the another reason.

Since we'll anyway need the retry I don't see why we should wait until the issues is closed. If we merge this PR in it will reduce the duration of the ducktape tests and we'll benefit from it on each test run.

Reply:
I guess I'm mixing an sorta-orthogonal concern in here: I worry that if we merge this and move on, we might forget about this bug in the client, and our customers will end up hitting it in their applications when they update to redpanda 22.1 and idempotency switches on by default.  This felt like a good point for us to dig in and fix it (even though we don't usually own these clients).

I agree that we will need the retries+timeouts anyway, but it seems like this specific step of destroying+recreating the client is going to go away once that upstream client issue is fixed.  If you feel strongly about the timeline for merging this PR, then I'm okay with merging it sooner -- would be nice to have a link in the code to the upstream issue though, so that when someone later wonders "why do we recreate the client?" they can figure it out.

Reply:
The confluent_python [issue](https://github.com/confluentinc/confluent-kafka-python/issues/1335) got traction and I added a comment explaining the retry in our code. Let's get it in and iterate when the upstream issue is fixed.